### PR TITLE
add support for Zarr datatype

### PR DESCRIPTION
**Pull request by annefou**
(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/datatypes/text.py [hunk 1]
```diff
@@ -84,6 +84,44 @@ def sniff_prefix(self, file_prefix: FilePrefix) -> bool:
...
+        if not dataset.dataset.purged:
+            dataset.peek = "Zarr file"
+            dataset.blurb = nice_size(dataset.get_size())
+        else:
+            dataset.peek = "file does not exist"
+            dataset.blurb = "file purged from disk"
+    
+    def get_mime(self) -> str:
+        """Returns the mime type of the datatype"""
+        return "text/html"
```
Comment (bgruening) [inline lib/galaxy/datatypes/text.py]: Does Zarr has a mimetype?

Comment (annefou) [reply lib/galaxy/datatypes/text.py]: I did not find anything specific mime type for zarr. In some documentation, the mime type was set to `application/zip` (see [pygeopi](https://docs.pygeoapi.io/en/stable/data-publishing/ogcapi-coverages.html)) but I don't think it makes much sense. Does it?

Comment (bgruening) [reply lib/galaxy/datatypes/text.py]: I guess then I would remove it. html is a wrong mimetype I think,

File: lib/galaxy/datatypes/text.py [hunk 2]
```diff
@@ -84,6 +84,44 @@ def sniff_prefix(self, file_prefix: FilePrefix) -> bool:
         return False
 
 
+@build_sniff_from_prefix
+class Zarr(Text):
+    """Class describing a zarr dataset"""
+    
+    edam_format = "format_3915"
+    file_ext = "zarr"
+
```
Comment (bgruening) [inline lib/galaxy/datatypes/text.py:93]: should we at least add a metadata element with a `version`? Similar to other datatypes?

File: lib/galaxy/datatypes/text.py [hunk 3]
```diff
@@ -84,6 +84,44 @@ def sniff_prefix(self, file_prefix: FilePrefix) -> bool:
...
+        >>> fname = get_test_fname( 'complete.bed' )
+        >>> Zarr().sniff( fname )
+        False
+        >>> fname = get_test_fname( 'test.zarr' )
+        >>> Zarr().sniff( fname )
+        True
+        """
+        headers = iter_headers(file_prefix, None)
+        for hdr in headers:
+            if hdr and hdr[0].lower().find("<html>") >= 0:
```
Comment (bgruening) [inline lib/galaxy/datatypes/text.py:116]: I don't understand why we are here searching for html?

File: lib/galaxy/datatypes/text.py (hunk 1 above)
Comment (annefou) [reply lib/galaxy/datatypes/text.py]: Can I remove the function get_mime completely?

File: lib/galaxy/datatypes/text.py (hunk 3 above)
Comment (annefou) [reply lib/galaxy/datatypes/text.py:116]: What I would like to do is to go through the folder and search for .z* files but I did not manage to do it. Can you help?

File: lib/galaxy/datatypes/text.py (hunk 1 above)
Comment (bgruening) [reply lib/galaxy/datatypes/text.py]: Sure. Please look at the other datatypes.

File: lib/galaxy/datatypes/text.py (hunk 3 above)
Comment (bgruening) [reply lib/galaxy/datatypes/text.py:116]: here is a bit of inspiration ....

> https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/datatypes/spaln.py#L181

**Pull request merged**